config/m4/sysdep.m4

```diff
@@ -96,12 +96,41 @@ AC_CHECK_DECLS([ethtool_cmd_speed, SPEED_UNKNOWN], [], [],
 
 
 #
-# PowerPC query for TB frequency
+# PowerPC "sys/platform/ppc.h" header
 #
-AC_CHECK_DECLS([__ppc_get_timebase_freq], [], [], [#include <sys/platform/ppc.h>])
 AC_CHECK_HEADERS([sys/platform/ppc.h])
 
 
+#
+# PowerPC query for getting TB frequency
+#
+AC_CHECK_DECL([__ppc_get_timebase_freq],
+              [AC_CHECK_FUNCS([__ppc_get_timebase_freq])], [],
+              [#include <sys/platform/ppc.h>])
+
+
+#
+# PowerPC query for getting TB.
+# Note: AC_CHECK_FUNCS doesn't work for checking __ppc_get_timebase()
+#
+AC_LINK_IFELSE([AC_LANG_SOURCE([[
+                #include <sys/platform/ppc.h>
+
+				int main(int argc, char** argv)
+                {
+					__ppc_get_timebase();
+					return 0;
+				} ]])],
+                [AC_MSG_RESULT([yes])
+                __ppc_get_timebase_happy=yes],
+                [AC_MSG_RESULT([no])
+                __ppc_get_timebase_happy=no])
+if test "x$__ppc_get_timebase_happy" == xyes; then
+    AC_DEFINE([HAVE___PPC_GET_TIMEBASE], [1],
+              [Define to 1 if you have the `__ppc_get_timebase' function.])
```

Comment:
can move this inside the AC_LINK_IFELSE?

Reply:
to avoid __ppc_get_timebase_happy var

Reply:
done